Add OpenSense Example Scripts and Data

jimmyDunne opened the pull request:
Adding Scripts and data for the OpenSense Example

### Brief summary of changes
- Edited large data files (22min) found in IMUData to smaller length (20secs)
- Added scripts and Matlab convenience classes
- updated Cmake to copy files during build

### Testing I've completed
- I have run all scripts and made sure that there are no errors, end-to-end. 
- Built on local system and confirmed that the files get written to;  install/share/doc/OpenSim/Code/Matlab/OpenSenseExampleFiles
- I have had exceptions using the command line and will perform further testing to identify and resolve. 

### Looking for feedback on...
- size of data files. 
- model location and use (rajagopal model)
- structure of helper classes (imuDataSlicer() and orientationTrackingHelper() ) 
- Any spelling or syntax improvements. 

### CHANGELOG.md (choose one)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/opensim-org/opensim-core/2466)
<!-- Reviewable:end -->

jimmyDunne commented:
Related to https://github.com/opensim-org/BananaPhone/issues/24

jenhicks commented on Bindings/Java/Matlab/OpenSenseExampleFiles/IMUDataConverter.m:
```diff
@@ -0,0 +1,61 @@
+
+%% Build an Xsens Settings Object. 
+% Instantiate the Reader Settings Class
+xsensSettings = XsensDataReaderSettings('MT_012005D6_009-001_Mappings.xml');
```
The name of the file included is imuMappings.xml
You might call it something like myImuMappings.xml to make it more clear that this can be changed.

jimmyDunne commented:
Made updates to the package that includes;
- Updated model; reverted to the original Rajagopal 2015 model then removed muscles and markers, edited geometry paths for femurs and tibias. 
- Added scripts for performing data conversion, calibration, and tracking without the use of convenience functions. 
- Added a utility folder that holds convenience classes and example scripts for using the convenience classes. 
- Updated all the path names in the scripts.
- Made updates to the documentation using the updated file names. 
- tested scripts and command line calls and confirmed consistency.

jimmyDunne commented:
- Updated from edits made by @aseth1. In particular, is the bone color. 
- Updated some paths and synced them with documentation. 

Otherwise, everything seems to be working well. 

I have added the example files to the [OpenSense SimTK download page.](https://simtk.org/frs/?group_id=1715).

jimmyDunne commented:
-Removed utility scripts. Will PR separately.

aseth1 commented:
___
*[Bindings/Java/Matlab/OpenSenseExampleFiles/OpenSense_OrientationTracking.m, line 37 at r5](https://reviewable.io/reviews/opensim-org/opensim-core/2466#-LgYVBMCCeXIhtKmoIw1:-LgYVBMCCeXIhtKmoIw2:b-jhh2wk) ([raw file](https://github.com/opensim-org/opensim-core/blob/20f6f6c6587d7375a6b94a72bcafe8adbd486361/Bindings/Java/Matlab/OpenSenseExampleFiles/OpenSense_OrientationTracking.m#L37)):*
> ```
> baseIMUName = 'pelvis_imu';     % The name of the base IMU to use for the model
> baseIMUHeading = 'z';      % The Coordinate Heading of the base IMU ('x', 'y', or 'z')
> visualizeTracking = 1;     % Boolean to Visualize the tracking simulation
> ```

Don't forget to make these consistent with changes to Calibration.


<!-- Sent from Reviewable.io -->

jenhicks commented:
I think it makes sense to use the Rajagopal model name. We can explain that in the documentation and add a link to her simtk page so people can find out more information about her model, if desired.

jimmyDunne commented:
I don't know what this 'reviewable' tool is or why we are using it but I won't be using it in the future. Why are we using yet another thing?

jimmyDunne commented:
Dealt with all comments.

aseth1 commented:
___
*[Bindings/Java/Matlab/OpenSenseExampleFiles/myIMUMappings.xml, line 9 at r5](https://reviewable.io/reviews/opensim-org/opensim-core/2466#-LgYR8qo-_quDGPhF4tc:-LgYoxso-QbpIuuGTBx-:bhipepe) ([raw file](https://github.com/opensim-org/opensim-core/blob/20f6f6c6587d7375a6b94a72bcafe8adbd486361/Bindings/Java/Matlab/OpenSenseExampleFiles/myIMUMappings.xml#L9)):*
<details><summary><i>Previously, jimmyDunne wrote…</i></summary><blockquote>

* this is generated from the tools so it needs to be fixed there too
</blockquote></details>

Good point. It'll be in the next PR I make or commit to.


<!-- Sent from Reviewable.io -->

jenhicks commented:
@jimmyDunne Did you change the name of the model where it is referenced on the confluence page? I think that is the only change that would affect the documentation.

jimmyDunne commented:
Yes, changed it on Confluence already.

jenhicks commented:
Can you send me a zip of the latest files? I will run through one more time on my machine, and then I think we're good to merge.

jimmyDunne commented:
Zip file attached here. 

[OpenSenseExampleFiles.zip](https://github.com/opensim-org/opensim-core/files/3254565/OpenSenseExampleFiles.zip)

jenhicks commented:
A couple small things came up related to changing the model name:

1) You need to change the name of the model in the osim (the xml tag model_name) to Rajagopal_2015. The model name is used to name the calibrated model. (It is currently imuTrackingModel, so the model that calibrate outputs is still imuTrackingModel) 

2) You also need to update the capture of the XML setup file for IK ... it shows the old model name.

I was able to run from command line and run the Matlab scripts without issue.

jimmyDunne commented:
Thanks. Those comments have been addressed.

jenhicks commented:
Looks good, thanks.
